hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SequenceIdGenerator.java

```diff
@@ -393,20 +398,35 @@ public static void upgradeToSequenceId(SCMMetadataStore scmMetadataStore)
           CONTAINER_ID, sequenceIdTable.get(CONTAINER_ID));
     }
 
-    // upgrade root certificate ID
-    if (sequenceIdTable.get(ROOT_CERTIFICATE_ID) == null) {
-      long largestRootCertId = BigInteger.ONE.longValueExact();
+    upgradeToCertificateSequenceId(scmMetadataStore, false);
+  }
+
+  public static void upgradeToCertificateSequenceId(
+      SCMMetadataStore scmMetadataStore, boolean force) throws IOException {
+    Table<String, Long> sequenceIdTable = scmMetadataStore.getSequenceIdTable();
+
+    // upgrade certificate ID table
+    if (sequenceIdTable.get(CERTIFICATE_ID) == null || force) {
+      // Start from ID 2.
+      // ID 1 - root certificate, ID 2 - first SCM certificate.
+      long largestCertId = BigInteger.ONE.add(BigInteger.ONE).longValueExact();
       try (TableIterator<BigInteger,
           ? extends KeyValue<BigInteger, X509Certificate>> iterator =
                scmMetadataStore.getValidSCMCertsTable().iterator()) {
         while (iterator.hasNext()) {
           X509Certificate cert = iterator.next().getValue();
-          if (HASecurityUtils.isSelfSignedCertificate(cert) &&
-              HASecurityUtils.isCACertificate(cert)) {
-            largestRootCertId =
-                Long.max(cert.getSerialNumber().longValueExact(),
-                    largestRootCertId);
-          }
+          largestCertId = Long.max(cert.getSerialNumber().longValueExact(),
+              largestCertId);
+        }
+      }
+
+      try (TableIterator<BigInteger,
+          ? extends KeyValue<BigInteger, X509Certificate>> iterator =
+               scmMetadataStore.getValidCertsTable().iterator()) {
+        while (iterator.hasNext()) {
+          X509Certificate cert = iterator.next().getValue();
+          largestCertId = Long.max(
```

Comment:
Is it possible here that we have an older certificate that still has an id out of the long range here?